cami/tools/linux/generate_yaml.py

```diff
@@ -544,78 +558,115 @@
 		"IntroName": "NsProxy",
 		"KernelName": "struct nsproxy",
 		"Fields": [
-			# {
-			# 	"IntroName": "CountOffset",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "UtsOffset",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "IpcOffset",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "MntOffset",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "PidOffset",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "NetOffset",
-			# 	"KernelName": "",
-			# },
+			{
+				"IntroName": "CountOffset",
+				"KernelName": "count",
+			},
+			{
+				"IntroName": "UtsOffset",
+				"KernelName": "uts_ns",
+			},
+			{
+				"IntroName": "IpcOffset",
+				"KernelName": "ipc_ns",
+			},
+			{
+				"IntroName": "MntOffset",
+				"KernelName": "mnt_ns",
+			},
+			{
+				"IntroName": "PidOffset",
+				"KernelName": "pid_ns_for_children",
+			},
+			{
+				"IntroName": "NetOffset",
+				"KernelName": "net_ns",
+			},
 		]
 	},
 	{
 		"IntroName": "Ungrouped",
-		"KernelName": "",
-		"Fields": [
-			# {
-			# 	"IntroName": "FileDentryOffset",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "ProtoNameOffset",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "SignalListHeadOffset",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "SocketAllocVfsInodeOffset",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "Running",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "FilePathOffset",
-			# 	"KernelName": "",
-			# },
-			# {
-			# 	"IntroName": "SignalNrThreadsOffset",
-			# 	"KernelName": "",
-			# },
-		]
-	},
+		"KernelName": None,
+        "Fields": [
+            {
+                "IntroName": "FileDentryOffset",
+                "Chain": [("struct file", "f_path"), ("struct path", "dentry")]
+            },
+            {
+                "IntroName": "ProtoNameOffset",
+                "Chain": [("struct proto", "name")]
+            },
+            {
+                "IntroName": "SignalListHeadOffset",
+                "Chain": [("struct signal_struct", "thread_head")]
+            },
+            {
+                "IntroName": "SocketAllocVfsInodeOffset",
+                "Chain": [("struct socket_alloc", "vfs_inode")]
+            },
+            {
+                "IntroName": "Running",
+                "Lambda": (lambda input_dict: input_dict["__SYSTEM_STATE_RUNNING__"])
+            },
+            {
+                "IntroName": "FilePathOffset",
+                "Chain": [("struct file", "f_path")]
+            },
+            {
+                "IntroName": "SignalNrThreadsOffset",
+                "Chain": [("struct signal_struct", "nr_threads")]
+            },
+
+        ]
+    },
 ]
 
-def create_bindings(input_dict):
+def get_bit_pos(input_dict, structure, bit):
+    # This is a little awkward because we assume the dictionary is sorted. No one can guarantee this. But it works.
```

Comment:
We shouldn't assume that the dictionary is sorted.

Reply:
At this moment gdb does not offer a reliable way to fetch the position of a bit in a bit field. While I do agree we should not make assumptions on undefined behaviours, this is the solution I found that works for this scenario.

Reply:
> At this moment gdb does not offer a reliable way to fetch the position of a bit in a bit field. While I do agree we should not make assumptions on undefined behaviours, this is the solution I found that works for this scenario.

You can do a little trick to make sure you have a sorted dictionary by first sorting the current items and then adding them to a `OrderedDict`, which preserves the order in which elements are added.